rust_snuba/src/processors/utils.rs

```diff
@@ -39,3 +40,38 @@ where
     };
     Ok(seconds_since_epoch.timestamp() as u32)
 }
+
+#[allow(dead_code)]
+pub struct CogsAccountant {
+    accountant: UsageAccountant<KafkaProducer>,
+}
+
+impl CogsAccountant {
+    #[allow(dead_code)]
+    fn new(bootstrap_servers: &str) -> Self {
+        let config = KafkaConfig::new_producer_config(bootstrap_servers, None);
+        Self {
+            accountant: UsageAccountant::new_with_kafka(config, None, None),
+        }
+    }
+
+    fn record_bytes(&mut self, resource_id: &str, app_feature: &str, amount_bytes: u64) {
+        if let Err(err) =
+            self.accountant
+                .record(resource_id, app_feature, amount_bytes, UsageUnit::Bytes)
+        {
+            tracing::warn!(?err, "error recording cogs");
```

Comment:
I think it would be a good idea to debounce this logs. 
Let's say that something goes wrong with the accountant. Not that there is much that can go wrong on the record method consistently, but anyway .... 
In that case you do not want to log a warning, potentially for each message you consume. The cogs accountant should never compromise the stability of the consumer itself.

Reply:
i feel that this pattern (and the fact that we use it aggressively at sentry) is a holdover from the trauma of writing performance sensitive software in Python, and particularly the fact that the GIL prevents us from offloading instrumentation overhead fully off to another thread.

we are also in a situation where "get the current system time" is one of the most expensive things we do on the main thread, so debouncing instead of just writing to stdout might harm performance.

empirically we know that it does not harm stability, because we did deploy with `println()` statements accidentally left in (per-poll, per-message) and also ran our experimental consumers with log-level=debug at some point

we should have the opportunity to question whether we are performing this debouncing at the right layer of abstraction. I feel that for printing logs, we already get the current system time there, and so ideally we'd be able to debouce in the log stdout handler if we really have to.

Reply:
Alright, I changed this to only log to sentry once. Once it has logged, it will never log again until the next deployment (or rebalance).

This way we do not harm performance by getting the current system time. At the same time, we do not entirely miss warnings. Once is enough to tell us if something is wrong.